### PR TITLE
Add test_source_waveform_parallel_site_unique in nidigital

### PR DESCRIPTION
**Pull request by sbethur**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~~- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~

- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

Add test_source_waveform_parallel_site_unique in nidigital, including disabled test cases for numpy.array and list input data types.

### List issues fixed by this Pull Request below, if any.

None

### What testing has been done?

Ran the test

### CODE REVIEW

**Comment by sbethur** [comment]
@mlopezleon, this test is for #1069, where we provided a way for users to specify the source waveforms as a Python dictionary (i.e. hashmap) of _sites --> waveform_data_.

File: src/nidigital/system_tests/test_system_nidigital.py [hunk 1]
```diff
@@ -123,6 +119,69 @@ def get_test_file_path(test_name, file_name):
...
+        elif source_waveform_type == numpy.array:
+            source_waveform = {
+                1: numpy.array([i for i in range(num_samples)]),
+                0: numpy.array([i for i in reversed(range(num_samples))])}
+        elif source_waveform_type == list:
+            source_waveform = {
+                1: [i for i in range(num_samples)],
+                0: [i for i in reversed(range(num_samples))]}
+        else:
+            assert False
```
Comment (mlopezleon) [inline src/nidigital/system_tests/test_system_nidigital.py]: Is there a way to include a message with the assertion to indicate why it happened?

File: src/nidigital/system_tests/test_system_nidigital.py [hunk 2]
```diff
@@ -123,6 +119,69 @@ def get_test_file_path(test_name, file_name):
...
+
+        multi_instrument_session.load_pattern(get_test_file_path(test_name, 'pattern.digipat'))
+
+        num_samples = 256
+        multi_instrument_session.write_sequencer_register(reg='reg0', value=num_samples)
+
+        multi_instrument_session.create_source_waveform_parallel(
+            pin_list='LowPins',
+            waveform_name='src_wfm',
+            data_mapping=2601)
```
Comment (mlopezleon) [inline src/nidigital/system_tests/test_system_nidigital.py]: Is data mapping going to turn into an enum?

File: src/nidigital/system_tests/test_system_nidigital.py (hunk 1 above)
Comment (sbethur) [reply src/nidigital/system_tests/test_system_nidigital.py]: Updated

File: src/nidigital/system_tests/test_system_nidigital.py (hunk 2 above)
Comment (sbethur) [reply src/nidigital/system_tests/test_system_nidigital.py]: Yes, it will done in #1066.

**Pull request merged**